src/erlazure_insights.erl

```diff
@@ -72,17 +72,28 @@ request_api(Url, Config) ->
 http_request(SubscriptionId, Filter, Select, Config) ->
     URL = "https://management.azure.com/subscriptions/" ++
     SubscriptionId ++
-    "/providers/microsoft.insights/eventtypes/management/values?" ++ "api-version=2014-04-01" ++ 
-    "&%24filter=" ++ Filter ++
-    "&%24select=" ++ Select,
+    "/providers/microsoft.insights/eventtypes/management/values?" ++
+    make_query_string([
+            {"api-version", "2014-04-01"},
+            {"$filter", Filter},
+            {"$select", Select}
+        ]),
     request_api(URL, Config).
 
+make_query_string(Params) ->
+    lists:foldl(fun({Key, Value}, Acc) -> 
+                case Value of
+                    [] -> Acc;
+                    _ -> Acc ++ "&" ++ Key ++ "=" ++ http_uri:encode(Value)
```

Comment:
if the first `{Key, Value}` pair will be `{Key, ""}`, then "&" will be added anyway, and the whole result will start with `&`:

```&key2=value2&key3=value3```